### ISSUE TITLE
linter: `oxc` incorrectly emits oxc(eslint(vars-on-top)) for `declare` block in TypeScript code

### ISSUE THREAD

**Issue by hamirmahal** [oxc-project/oxc#14951]
### What version of Oxlint are you using?

1.18.0

### What command did you run?

npx oxlint

### What does your `.oxlintrc.json` config file look like?

```jsonc
{
	"categories": {
		"style": "deny"
	}
}
```

### What happened?

`oxc` emits oxc([eslint(vars-on-top)](https://oxc.rs/docs/guide/usage/linter/rules/eslint/vars-on-top.html)) for

```ts
declare global {
	var __CUSTOM_FLAG__: boolean | undefined;
}
```

but this needs to be `var`.

---
If we use `let` instead,
```ts
declare global {
	let __CUSTOM_FLAG__: boolean | undefined;
}
```
TypeScript reports
```
Property '__CUSTOM_FLAG__' does not exist on type 'typeof globalThis'.ts(2339)
```
when accessing
```ts
globalThis.__CUSTOM_FLAG__
```
